contracts/cw20-base/src/allowances.rs

```diff
@@ -20,9 +20,12 @@ pub fn execute_increase_allowance(
         return Err(ContractError::CannotSetOwnAccount {});
     }
 
-    let update_fn = |allow: Option<AllowanceResponse>| -> StdResult<_> {
+    let update_fn = |allow: Option<AllowanceResponse>| -> Result<_, _> {
         let mut val = allow.unwrap_or_default();
         if let Some(exp) = expires {
+            if exp.is_expired(&env.block) {
+                return Err(ContractError::Expired {});
```

Comment:
More a suggestion than a blocker. I think what we want is more a validation error, just to be clearer to the user/consumer. The problem isn't that an existing allowance is expired, the problem is the user is trying to create/modify an allowance so that it expires in the past.

I'd add a new error variant here:
https://github.com/CosmWasm/cw-plus/blob/main/contracts/cw20-base/src/error.rs